src/Compilers/CSharp/Portable/Symbols/ObsoleteAttributeHelpers.cs

```diff
@@ -123,6 +147,30 @@ internal static ObsoleteDiagnosticKind GetObsoleteDiagnosticKind(Symbol symbol,
                     // later stage
                     return ObsoleteDiagnosticKind.LazyPotentiallySuppressed;
             }
+
+            static bool isExperimentalSymbol(NamedTypeSymbol namedTypeSymbol)
+            {
+                return namedTypeSymbol.Arity == 0
+                    && namedTypeSymbol.HasNameQualifier("System.Diagnostics.CodeAnalysis")
+                    && namedTypeSymbol.Name.Equals("ExperimentalAttribute", StringComparison.Ordinal);
```

Comment:
:memo: This is a poor man's cycle avoidance solution. This would treat any type of the right shape as an `ExperimentalAttribute`, not merely *the* well-known one...